modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeCommerceShippingOptionCheck.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class JavaUpgradeCommerceShippingOptionCheck extends BaseJavaTermCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, JavaTerm javaTerm,
+			String fileContent)
+		throws Exception {
+
+		List<String> importNames = javaTerm.getImportNames();
+
+		String javaTermContent = javaTerm.getContent();
+
+		if (!importNames.contains(
+				"com.liferay.commerce.model.CommerceShippingOption")) {
+
+			return javaTermContent;
+		}
+
+		Pattern pattern = _getPattern();
+
+		Matcher matcher = pattern.matcher(javaTermContent);
+
+		while (matcher.find()) {
+			List<String> parameterList = JavaSourceUtil.getParameterList(
+				matcher.group(1));
+
+			if (parameterList.size() == 3) {
+				String newInstance = StringUtil.replace(
+					matcher.group(), matcher.group(2),
+					_getNewInstance(parameterList));
+
+				javaTermContent = StringUtil.replace(
+					javaTermContent, matcher.group(), newInstance);
+			}
+		}
+
+		return javaTermContent;
+	}
+
+	@Override
+	protected String[] getCheckableJavaTermNames() {
+		return new String[] {JAVA_METHOD};
+	}
+
+	private String _getNewInstance(List<String> parameters) {
```

Comment:
@michaelmdelcavalcanti, could we rename this method to `_formatParameters` or `_getNewParameters` to clarify what the function does?